### PR TITLE
Optimize profiler, bump limit

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**

### CODE REVIEW

**Review by kasperski95 — APPROVED** [review]
LGTM

File: protostar/profiler/contract_profiler.py [hunk 1]
```diff
@@ -179,15 +179,12 @@ def create_instruction_list(
     return instructions
 
 
-def find_instruction(instructions: list[Instruction], pc: Address) -> Instruction:
-    for instr in instructions:
-        if instr.pc == pc:
-            return instr
-    assert False
+def find_instruction(instructions: dict[int, Instruction], pc: Address) -> Instruction:
+    return instructions[pc]
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply protostar/profiler/contract_profiler.py:183]: I don't want to, I want to keep it abstracted for now, I may want to introduce different algorithm

[the remark this answers is not in the document]
Comment (mkaput) [reply protostar/profiler/contract_profiler.py:183]: And actually I think you might consider using a `UserDict` to abstract the whole concept of storing Instructions in query-efficient manner

File: protostar/profiler/contract_profiler.py [hunk 2]
```diff
@@ -179,15 +179,12 @@ def create_instruction_list(
...
 
 
 def build_call_callstacks(
-    instructions: list[Instruction], tracer_data: TracerDataManager
+    instructions: dict[int, Instruction], tracer_data: TracerDataManager
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply protostar/profiler/contract_profiler.py]: not relevant anymore, this is now a class with name `Instructions`

**Pull request merged**